lib/garage_client/client.rb

```diff
@@ -74,6 +74,14 @@ def connection
       end
     end
 
+    def request_options
+      options[:request] || default_options.options[:request]
+    end
+
+    def request_options=(value)
+      options[:request] = value
+    end
+
```

Comment:
It seems these methods are similar to `headers` and `headers=` methods. Please move under `headers` and `headers=` definition if so.